Remotion/Data/DomainObjects/Linq/ScalarResultRowAdapter.cs

```diff
@@ -48,6 +48,7 @@ public IStorageTypeInformationProvider StorageTypeInformationProvider
       get { return _storageTypeInformationProvider; }
     }
 
+    [return: MaybeNull]
     public T GetValue<T> (ColumnID columnID)
```

Comment:
attribute vs `?`

Reply:
Intentional. the `?` doesn't make a not-nullable value type nullable, you still need to use a nullable value type for `T`. Thus, I went with the attribute variant with the intention of better showing that T is taken literally, but null is still a possiblity, so please be aware. The `GetValue<T>` method is one of the big iffy spots. There's just no great way to do this, I think.

Reply:
That `?` does not apply for value types is something one has to know, I don't think the intention is clearer by applying the attribute, is its just the same thing in another format - but thats just my opinion.

Reply:
I understand the opinion and was in the same situation. I went first with the `?` and then noticed a mistake in the unit tests. I think we can (and should revisit this some time down the road) but at the beginning, devs who read the API but don't know NRT inside out will not find a false friend (when compared to nullable value types) in the `?`. That's why I chose to stick with `MaybeNull`.

So, I think it's easier to not step into a potential bug when you're just beginning with NRT as an application developer.

Reply:
I can't agree but it's alright for me, I'll approve the PR then.